Tinder.DAL/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Tinder.DAL.Interfaces;
+using Tinder.DAL.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Tinder.DAL.Repositories
+{
+    public class GenericRepository<TEntity> : IGenericRepository<TEntity> where TEntity : BaseEntity
+    {
+        protected ApplicationDbContext _context;
+        public GenericRepository(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+        public virtual async Task<TEntity> CreateAsync(TEntity entity, CancellationToken cancellationToken)
+        {
+           await _context.Set<TEntity>().AddAsync(entity, cancellationToken);
+           await _context.SaveChangesAsync(cancellationToken);
+           return entity;
+        }
+
+        public virtual async Task<IEnumerable<TEntity>> GetAllAsync(CancellationToken cancellationToken)

```

Comment:
async/await can be removed. Also can be transformed to the expression method

Reply:
public virtual Task<IEnumerable<TEntity>> GetAllAsync(CancellationToken cancellationToken) => 
         _context.Set<TEntity>()
                .AsNoTracking()
                .ToListAsync(cancellationToken)